File: src/interfaces/java/src/org/brlcad/geometryservice/net/msg/SessionInfoMsg.java
/*
 * BRL-CAD
 *
 * Copyright (c) 2011 United States Government as represented by
 * the U.S. Army Research Laboratory.
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public License
 * version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this file; see the file named COPYING for more
 * information.
 */
/** @file SessionInfo.java
 * 
 */
package org.brlcad.geometryservice.net.msg;

import java.util.UUID;

import org.brlcad.geometryservice.net.ByteBufferReader;
import org.brlcad.geometryservice.net.ByteBufferWriter;

/**
 * @author david.h.loman
 * 
 */
public class SessionInfoMsg extends AbstractNetMsg {

	private UUID sessionID;

	/**
	 * @param msgType
	 */
	public SessionInfoMsg(UUID sessionID) {
		super(NetMsgTypes.NewSessionREQ);
		this.sessionID = sessionID;
	}

	/**
	 * @param msgType
	 * @param reMsg
	 */
	public SessionInfoMsg(AbstractNetMsg reMsg, UUID sessionID) {
		super(NetMsgTypes.NewSessionREQ, reMsg);
		this.sessionID = sessionID;
	}

	/**
	 * @param msgType
	 * @param reader
	 */
	public SessionInfoMsg(ByteBufferReader reader) {
		super(NetMsgTypes.NewSessionREQ, reader);
	}

	/**
	 * @see org.brlcad.geometryservice.net.msg.AbstractNetMsg#_deserialize(org.brlcad.geometryservice.net.ByteBufferReader)
	 */
	@Override
	protected void _deserialize(ByteBufferReader reader) {
		this.sessionID = reader.getUUID();
	}

	/**
	 * @see org.brlcad.geometryservice.net.msg.AbstractNetMsg#_serialize(org.brlcad.geometryservice.net.ByteBufferWriter)
	 */
	@Override
	protected void _serialize(ByteBufferWriter writer) {
		writer.putUUID(this.sessionID);
	}

	/**
	 * @return the sessionID
	 */
	public final UUID getSessionID() {
		return sessionID;
	}

}

File: src/libNet/netMsg/SessionInfoMsg.cxx
/*            S E S S I O N I N F O M S G . C X X
 * BRLCAD
 *
 * Copyright (c) 2011 United States Government as represented by
 * the U.S. Army Research Laboratory.
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public License
 * version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this file; see the file named COPYING for more
 * information.
 */
/** @file SessionInfoMsg.cxx
 *
 * Brief description
 *
 */

#include "SessionInfoMsg.h"

/* Normal Constructor */
SessionInfoMsg::SessionInfoMsg(GSUuid* sessionID) :
    NetMsg(SESSIONINFO), sessionID(sessionID)
{}

/* Reply Constructor */
SessionInfoMsg::SessionInfoMsg(NetMsg* msg, GSUuid* sessionID) :
	NetMsg(SESSIONINFO, msg), sessionID(sessionID)
{}

/* Deserializing Constructor */
SessionInfoMsg::SessionInfoMsg(ByteBuffer* bb, Portal* origin) :
    NetMsg(bb, origin)
{
    this->sessionID = new GSUuid(bb->getString()); //TODO modify GSUuid or ByteBuffer to streamline this
}

/* Destructor */
SessionInfoMsg::~SessionInfoMsg()
{}

bool
SessionInfoMsg::_serialize(ByteBuffer* bb)
{
  bb->putString(*this->sessionID->toString());
  return true;
}

std::string
SessionInfoMsg::toString()
{
  std::string out;

  out.append(NetMsg::toString());
  out.append("\t  SessionID: ");
  out.append(*this->sessionID->toString());

  return out;
}

bool
SessionInfoMsg::_equals(const NetMsg& msg)
{
  SessionInfoMsg& gmsg = (SessionInfoMsg&) msg;

  if (!this->getSessionID()->equals(gmsg.getSessionID()))
    return false;
  return true;
}

/*
 *Getters n Setters
 */
GSUuid*
SessionInfoMsg::getSessionID()
{
  return this->sessionID;
}

/*
 * Local Variables:
 * mode: C
 * tab-width: 8
 * indent-tabs-mode: t
 * c-file-style: "stroustrup"
 * End:
 * ex: shiftwidth=4 tabstop=8
 */
